### PR TITLE
Fixes content bug in Deutsch-Jozsa Algorithm

### LINKED ISSUE #1214: Deutsch-Jozsa Algorithm: Sec 2. Worked example
## Describe the Issue
1. The function corresponding to the specified oracle has not been provided. I believe that the function in question is f(x_0,x_1) = x_0 \oplus x_1 (see the attached note DJ_oracle)
2. A check that the proposed circuit indeed
[DJ_oracle.pdf](https://github.com/qiskit-community/qiskit-textbook/files/6832689/DJ_oracle.pdf)
 implements the corresponding phase oracle is not provided. (Again, see the attached note  DJ_oracle.)
3. Qiskit conventions for labeling qubits is not respected, either in the formulas or in the corresponding circuit diagram.

## Expected Solution
See attached note.

### PR DESCRIPTION
**Pull request by divshacker**
# Changes made

fixes #1214 

# Justification

I have added the oracle function and there truth table for more clarity how the oracle works. Also i have added the phase oracle as it was not mentioned in worked example. I have corrected the convention problem (which was previously **123**) to **012**

### CODE REVIEW

**Comment by divshacker** [comment]
Hey @nepomechie and @frankharkins I have created a PR. I have added few things (will add more) like using the oracle function and a phase function also I have corrected the convention problem. But as I said there are some things which need to be added further like more explanation regarding oracle function or phase oracle. I am get few suggestions regarding that.

**Comment by divshacker** [comment]
@nepomechie Please check the changes made and let me know if any further changes required.

**Review by frankharkins — COMMENTED** [review]
Looks good! I would make a couple of changes, but up to you and @nepomechie

File: content/ch-algorithms/deutsch-jozsa.ipynb [hunk 1]
```diff
@@ -195,50 +195,65 @@
     "\n",
     "Let's go through a specific example for a two bit balanced function: \n",
     "\n",
+    "Consider a two-bit function $f(x_0,x_1)=x_0 \\oplus x_1$ such that \n",
```
[the remark this answers is not in the document]
Comment (divshacker) [reply content/ch-algorithms/deutsch-jozsa.ipynb:198]: Didn't know about that. I followed paper(pdf file) and the function given in issue. I do agree with you as the x_1,x_0 is correct convention maybe @nepomechie would have some specific reason for writing this way?

File: content/ch-algorithms/deutsch-jozsa.ipynb [hunk 2]
```diff
@@ -195,50 +195,65 @@
...
-    "                  + \\lvert 1 0 \\rangle_{12} \\otimes \\left( \\lvert 0 \\oplus 1 \\oplus 0 \\rangle - \\lvert 1 \\oplus 1 \\oplus 0 \\rangle \\right)_{3} \\\\\n",
-    "                  + \\lvert 1 1 \\rangle_{12} \\otimes \\left( \\lvert 0 \\oplus 1 \\oplus 1 \\rangle - \\lvert 1 \\oplus 1 \\oplus 1 \\rangle \\right)_{3} \\right]\n",
+    "            \\lvert \\psi_2 \\rangle =  \\frac{1}{2\\sqrt{2}} \\left[ \\lvert 0 0 \\rangle_{01} \\otimes \\left( \\lvert 0 \\oplus 0 \\oplus 0 \\rangle - \\lvert 1 \\oplus 0 \\oplus 0 \\rangle \\right)_{2} \\\\\n",
+    "                  + \\lvert 0 1 \\rangle_{01} \\otimes \\left( \\lvert 0 \\oplus 0 \\oplus 1 \\rangle - \\lvert 1 \\oplus 0 \\oplus 1 \\rangle \\right)_{2} \\\\\n",
+    "                  + \\lvert 1 0 \\rangle_{01} \\otimes \\left( \\lvert 0 \\oplus 1 \\oplus 0 \\rangle - \\lvert 1 \\oplus 1 \\oplus 0 \\rangle \\right)_{2} \\\\\n",
+    "                  + \\lvert 1 1 \\rangle_{01} \\otimes \\left( \\lvert 0 \\oplus 1 \\oplus 1 \\rangle - \\lvert 1 \\oplus 1 \\oplus 1 \\rangle \\right)_{2} \\right]\n",
     "        \\end{align*}\n",
-    "        $$\n",
+    "         $$\n",
```
Comment (divshacker) [inline content/ch-algorithms/deutsch-jozsa.ipynb:240]: This equation is not rendering properly on github but working on local machine. I tried to fix it but no sucess. Can you give it a try ? @frankharkins

Comment (frankharkins) [reply content/ch-algorithms/deutsch-jozsa.ipynb:240]: Notebooks on github are a bit awkward, I think it's fine.

Comment (divshacker) [reply content/ch-algorithms/deutsch-jozsa.ipynb:240]: Ok great

File: content/ch-algorithms/deutsch-jozsa.ipynb (hunk 1 above)
[the remark this answers is not in the document]
Comment (frankharkins) [reply content/ch-algorithms/deutsch-jozsa.ipynb:198]: No reply so let's leave it as it is

**Review by frankharkins — APPROVED** [review]
Thank you!

**Pull request merged**